addons/account_bank_statement_import/res_partner_bank.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#     This file is part of account_bank_statement_import,
+#     an Odoo module.
+#
+#     Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)
+#
+#     account_bank_statement_importis free software:
+#     you can redistribute it and/or modify it under the terms of the GNU
+#     Affero General Public License as published by the Free Software
+#     Foundation,either version 3 of the License, or (at your option) any
+#     later version.
+#
+#     account_bank_statement_import is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with account_bank_statement_import_coda.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import re
+from openerp import api, models, fields
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    sanitized_acc_number = fields.Char(
```

Comment:
create and write are already overrided in base_iban. https://github.com/odoo/odoo/blob/master/addons/base_iban/base_iban.py#L83 In base_iban, the acc_numer is pretty formatted according to a standard IBAN repr.  I prefer to keep the acc_number untouched since it's used in some generated documents where the format is important.
I also sanitize the right operand of the search to make the search insensitive to the input format. (for example when searching in the UI)